framework/src/argument.rs

```diff
@@ -61,27 +66,36 @@ where
 /// the first segment is taken and parsed into an argument. If parsing succeeds,
 /// `Ok(Some(...))` is returned, otherwise `Err(...)`. The error is wrapped in
 /// [`ArgumentError::Argument`].
-pub fn optional_argument<T>(
+pub async fn optional_argument<T>(
+    ctx: &Context,
+    msg: &Message,
     segments: &mut ArgumentSegments<'_>,
-) -> Result<Option<T>, ArgumentError<T::Err>>
+) -> Option<Result<T, ArgumentError<T::Err>>>
```

Comment:
This return type is wrong. If a user defines an optional argument, the `#[command]` macro will generate code that will fail to compile. The return type of command functions is a `Result`, and you cannot use the `?` operator with an `Option` inside of a function returning a `Result`. Even if the operator did work, the user would have a `Result<T, ArgumentError<T::Err>>` instead of the `Option<T>` that they expect.

This should be changed back to the original return type, as it is intentional.